### PR TITLE
Add js sample

### PR DESCRIPTION
**Pull request by naoki-akimoto**
I added javascript/typescript sample. Please review.

### CODE REVIEW

**Comment by yongpingchen** [comment]
please add package.json and add thing-if dependence.

**Review by yongpingchen — COMMENTED** [review]
1. remove js file. Since it couldn't be check by the code, and can simply use the js compiled from ts file. 
2. use undefined instead of null for nullable parameters

File: snippets/js/generate_js.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
```
Comment (yongpingchen) [inline snippets/js/generate_js.sh:2]: add
```shell
typings install
```

File: snippets/js/generate_js.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+tsc
+js-beautify -s 2 src.js > tmp.js
```
Comment (yongpingchen) [inline snippets/js/generate_js.sh]: failed on js-beautify(v1.5.4), use `-o` instead
```shell
js-beautify -s 2 src.js -o tmp.js
```

**Review by yongpingchen — COMMENTED** [review]
left comments

**Comment by naoki-akimoto** [comment]
@yongpingchen fixed. Please confirm.

**Review by yongpingchen — APPROVED** [review]

**Pull request merged**